frame/evm-system/src/lib.rs

```diff
@@ -165,28 +176,47 @@ impl<T: Config> Pallet<T> {
 
 	/// Create an account.
 	pub fn create_account(who: &<T as Config>::AccountId) -> AccountCreationOutcome {
-		if Self::account_exists(who) {
-			return AccountCreationOutcome::AlreadyExists;
-		}
-
-		Account::<T>::insert(who.clone(), AccountInfo::<_, _>::default());
-		Self::on_created_account(who.clone());
-		AccountCreationOutcome::Created
+		Account::<T>::mutate_exists(who, |maybe_account| match maybe_account {
```

Comment:
Looks like what this code does is it introduces a counter for how many times an account was created; what's the idea here? Do we just increment the counter every time the `create_account` is called?

Why is this behaviour correct? Surely, with two `create_account` calls, the account won't be removed after the first time it's balance reaches zero - but it surely will be removed the second time.

The rule we should follow, I think, is whether an account has a code or not.